wdl-cli/src/analysis/results.rs

```diff
@@ -0,0 +1,107 @@
+//! Results of an analysis.
+
+use std::rc::Rc;
+
+use anyhow::Result;
+use anyhow::anyhow;
+use anyhow::bail;
+use wdl_analysis::AnalysisResult;
+use wdl_ast::AstNode as _;
+use wdl_ast::Diagnostic;
+
+use crate::analysis::Source;
+
+/// A set of analysis results.
+///
+/// If successfully created, the set of analysis results are guaranteed not to
+/// have any associated errors (but they may contain diagnostics).
+#[derive(Debug)]
+pub struct AnalysisResults(Vec<AnalysisResult>);
+
+impl AnalysisResults {
+    /// Attempts to create a new set of analysis results.
+    ///
+    /// Returns an error if an error was encountered during analysis. That being
+    /// said, each analysis result may have diagnostics.
+    pub fn try_new(results: Vec<AnalysisResult>) -> Result<Self> {
+        for result in results.iter() {
+            if let Some(e) = result.error() {
+                bail!(e.to_owned());
```

Comment:
I'm not sure this is the best behavior. We've done a full analysis at this point, so it's not like bailing on the _first_ error saves any compute. I think it would be better to return the full set of errors. We can leave it up to the caller whether the full set of errors is reported or just the first one.

Reply:
FWIW, this is just copied from the current `wdl` binary does (https://github.com/stjude-rust-labs/wdl/blob/0682dca08692717361b23e864276feb562d1a57e/wdl/src/bin/wdl.rs#L108-L110 and https://github.com/stjude-rust-labs/wdl/blob/0682dca08692717361b23e864276feb562d1a57e/wdl/src/bin/wdl.rs#L357-L359). That being said, since I'm formalizing it, I don't mind making it a new return type.